### PR TITLE
Fix `decodeETHData`

### PR DESCRIPTION
**Pull request by Ferret-san**
## Overview

`op_node` was returning an error in the `decodeETHData` function when attempting to read the `index` from the data. The error 
was an `unexpected EOF` when reading data from a buffer.

The error specifically occurred because because the buffer has a size of 12 bytes, 8 of which would be written into `height`, leaving 4 bytes for `index`, but since index was of type int64, there where not enough bytes in the buffer to write into it, thus causing the `unexpected EOF`.

The solution to this is to make `index` an `int32` variable.

### CODE REVIEW

**Comment by Ferret-san** [comment]
note that there are still errors in the op-node
image

File: op-node/rollup/derive/calldata_source.go [hunk 1]
```diff
@@ -143,14 +143,15 @@ func DataFromEVMTransactions(config *rollup.Config, daCfg *rollup.DAConfig, batc
 // decodeETHData will decode the data retrieved from the EVM, this data
 // was previously posted from op-batcher and contains the block height
 // with transaction index of the SubmitPFD transaction to the DA.
-func decodeETHData(celestiaData []byte) (int64, int64, error) {
+func decodeETHData(celestiaData []byte) (int64, int32, error) {
```
Comment (tuxcanfly) [inline op-node/rollup/derive/calldata_source.go:146]: This should match batcher, which is serializing as `int64, uint32`.

https://github.com/celestiaorg/optimism/pull/3/files#diff-1fc10714a3fe0349447d62b700894474ab8786e6d28bbeea56d017bfd137b676R171

Comment (tuxcanfly) [inline op-node/rollup/derive/calldata_source.go:146]:
```suggestion
func decodeETHData(celestiaData []byte) (int64, uint32, error) {
```

File: op-node/rollup/derive/calldata_source.go [hunk 2]
```diff
@@ -143,14 +143,15 @@ func DataFromEVMTransactions(config *rollup.Config, daCfg *rollup.DAConfig, batc
...
 	buf := bytes.NewBuffer(celestiaData)
 	var height int64
 	err := binary.Read(buf, binary.BigEndian, &height)
 	if err != nil {
 		return 0, 0, fmt.Errorf("error deserializing height: %w", err)
 	}
-	var index int64
+
+	var index int32
```
Comment (tuxcanfly) [inline op-node/rollup/derive/calldata_source.go:154]:
```suggestion
	var index uint32
```

**Review by tuxcanfly — APPROVED** [review]
OK.

**Comment by tuxcanfly** [comment]
Pulled into demo branch: https://github.com/ethereum-optimism/optimism/commit/82d4fee590cab5ee548de3c540da2b40203fe1f5 Will keep this open to merge into `celestia-develop` when concurrent batch submission is fixed.

**Review by tuxcanfly — COMMENTED** [review]
Needs to be rebased on top of `celestia-develop` as per latest workflow.

**Comment by jcstein** [comment]
Resolves https://github.com/celestiaorg/optimism/issues/13

**Pull request closed without merging**